### DISCUSSION TITLE
Can't connect to container services in container or host mode

### DISCUSSION THREAD

**Discussion by yapith-suplari** [community/community#27159] [Actions]
<p>Hi there,</p>
<p>I tried to set up a service container of localstack and postgres following the docs: <a href="https://docs.github.com/en/free-pro-team@latest/actions/guides/about-service-containers" class="inline-onebox" rel="noopener nofollow ugc">About service containers - GitHub Docs</a></p>
<p>I keep running into connection refused error connecting via docker network alias (in containerized job).</p>
<p>In host-mode I got further with a test non-containerized step, the client is able to establish a connection, but immediately terminated. This is odd because I couldn’t repro this issue on my localbox with the same localstack version and config.</p>
<p>I’m hoping someone could point out something wrong with my setup. I’m interested in getting containerized steps to be able to connect via network aliases, e.g. “<a href="http://localstack:4572" rel="noopener nofollow ugc">http://localstack:4572</a>”.</p>
<pre><code class="lang-auto">name: build

on:
  pull_request:
    branches: [ main ]

  workflow_dispatch:

jobs:
  build:
    runs-on: ubuntu-latest

    services:
      postgres:
        image: postgres:12.5
        env:
          POSTGRES_USER: $USER
        ports:
          - 5432:5432

      localstack:
        image: localstack/localstack:0.11.2
        env:
          SERVICES: ${SERVICES-s3}
          AWS_ACCESS_KEY_ID: test
          AWS_SECRET_ACCESS_KEY: test
          DEFAULT_REGION: us-west-2
        ports:
          - 4572:4572

    steps:
      - uses: actions/checkout@v2

# This step tries to connect to http://localstack:4572
      - name: Test
        uses: VaultVulp/action-pipenv@v2.0.1
        with:
          command: run python -m pytest --disable-pytest-warnings
        env:
          AWS_ACCESS_KEY_ID: test
          AWS_SECRET_ACCESS_KEY: test
          AWS_REGION: us-west-2
          DB_HOST: postgres
          S3_HOST: localstack
</code></pre>
<p>I also tried containerized job mode with a <code>container:</code> field set and it made no difference.</p>
<p>In containerized mode, I also tried taking out the <code>ports</code> mapping, per docs saying they’re not needed.</p>
<p>I also eliminated the possibility localstack just wasn’t ready for taking requests using a long sleep.</p>
<p>Thanks in advance.</p>
<p>hussein</p>

**Comment by yapith-suplari** [comment, marked as answer]
<p>Argh. Turns out it wasn’t a network issue. The network aliases work as advertised. I didn’t realize ${VAR-DEFAULT_VAL} is a bash syntax and not supported.</p>
<p>Closing the issue.</p>

**Reply by heshaam-42c** [reply]
Can you explain this a little more, I'm having the same issue and tried everything
